### ISSUE TITLE
Akka incorrectly schedules continuations after .Ask, causing deadlocks and/or delays

### ISSUE THREAD

**Issue by 0x53A** [akkadotnet/akka.net#2546]
__Edit3: fixed the repro and root cause, with possible fix__

The way Akka currently schedules continuations on Ask (and probably other places), user code runs on (and blocks) the Helios worker thread.

For one repro, see https://github.com/akkadotnet/akka.net/pull/2548/files , but it is NOT directly related to .Result, doing anything after ``await Ask`` locks up the system for the duration.

One possible fix would be to construct the ``TaskCompletionSource`` with ``TaskCreationOptions.RunContinuationsAsynchronously``. That way, the continuations (the user code) is executed on the thread pool, freeing the Helios worker thread.

see the callstack at this breakpoint:
![image](https://cloud.githubusercontent.com/assets/4236651/23656431/fa4d7806-0339-11e7-9564-97456ffae9ba.png)

```

>	AkkaDeadlockTest.exe!AkkaDeadlockTest.MainWindow.Button_Click_2(object sender, System.Windows.RoutedEventArgs e) Line 21	C#
 	[Resuming Async Method]	
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(object stateMachine)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.OutputAsyncCausalityEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.TaskAwaiter.OutputWaitEtwEvents.AnonymousMethod__0()	Unknown
 	mscorlib.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore.ContinuationWrapper.Invoke()	Unknown
 	mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Action action, bool allowInlining, ref System.Threading.Tasks.Task currentTask)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.__Canon>.TrySetResult(System.__Canon result)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskCompletionSource<System.__Canon>.TrySetResult(System.__Canon result)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskCompletionSource<System.__Canon>.SetResult(System.__Canon result)	Unknown
 	Akka.dll!Akka.Util.Internal.TaskExtensions.CastTask.AnonymousMethod__0(System.Threading.Tasks.Task<object> _)	Unknown
 	mscorlib.dll!System.Threading.Tasks.ContinuationTaskFromResultTask<object>.InnerInvoke()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.Execute()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecutionContextCallback(object obj)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteWithThreadLocal(ref System.Threading.Tasks.Task currentTaskSlot)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.ExecuteEntry(bool bPreventDoubleExecution)	Unknown
 	mscorlib.dll!System.Threading.Tasks.ThreadPoolTaskScheduler.TryExecuteTaskInline(System.Threading.Tasks.Task task, bool taskWasPreviouslyQueued)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskScheduler.TryRunInline(System.Threading.Tasks.Task task, bool taskWasPreviouslyQueued)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskContinuation.InlineIfPossibleOrElseQueue(System.Threading.Tasks.Task task, bool needsProtection)	Unknown
 	mscorlib.dll!System.Threading.Tasks.StandardTaskContinuation.Run(System.Threading.Tasks.Task completedTask, bool bCanInlineContinuationTask)	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task.FinishStageThree()	Unknown
 	mscorlib.dll!System.Threading.Tasks.Task<System.__Canon>.TrySetResult(System.__Canon result)	Unknown
 	mscorlib.dll!System.Threading.Tasks.TaskCompletionSource<System.__Canon>.TrySetResult(System.__Canon result)	Unknown
 	Akka.dll!Akka.Actor.FutureActorRef.TellInternal(object message, Akka.Actor.IActorRef sender)	Unknown
 	Akka.dll!Akka.Actor.ActorRefBase.Tell(object message, Akka.Actor.IActorRef sender)	Unknown
 	Akka.Remote.dll!Akka.Remote.DefaultMessageDispatcher.Dispatch(Akka.Actor.IInternalActorRef recipient, Akka.Actor.Address recipientAddress, SerializedMessage message, Akka.Actor.IActorRef senderOption)	Unknown
 	Akka.Remote.dll!Akka.Remote.EndpointReader.Reading.AnonymousMethod__11_1(Akka.Remote.Transport.InboundPayload inbound)	Unknown
 	[Lightweight Function]	
 	Akka.dll!Akka.Tools.MatchHandler.PartialHandlerArgumentsCapture<System.__Canon, System.__Canon, System.__Canon, System.__Canon>.Handle(System.__Canon value)	Unknown
 	Akka.dll!Akka.Actor.ReceiveActor.ExecutePartialMessageHandler(object message, Akka.Tools.MatchHandler.PartialAction<object> partialAction)	Unknown
 	Akka.dll!Akka.Actor.ReceiveActor.OnReceive(object message)	Unknown
 	Akka.dll!Akka.Actor.UntypedActor.Receive(object message)	Unknown
 	Akka.dll!Akka.Actor.ActorBase.AroundReceive(Akka.Actor.Receive receive, object message)	Unknown
 	Akka.dll!Akka.Actor.ActorCell.ReceiveMessage(object message)	Unknown
 	Akka.dll!Akka.Actor.ActorCell.Invoke(Akka.Actor.Envelope envelope)	Unknown
 	Akka.dll!Akka.Dispatch.Mailbox.ProcessMailbox(int left, long deadlineTicks)	Unknown
 	Akka.dll!Akka.Dispatch.Mailbox.ProcessMailbox()	Unknown
 	Akka.dll!Akka.Dispatch.Mailbox.Run.AnonymousMethod__36_0()	Unknown
 	Akka.dll!Akka.Actor.ActorCell.UseThreadContext(System.Action action)	Unknown
 	Akka.dll!Akka.Dispatch.Mailbox.Run()	Unknown
 	Akka.dll!Helios.Concurrency.DedicatedThreadPool.PoolWorker.RunThread()	Unknown
 	mscorlib.dll!System.Threading.ThreadHelper.ThreadStart_Context(object state)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state, bool preserveSyncCtx)	Unknown
 	mscorlib.dll!System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state)	Unknown
 	mscorlib.dll!System.Threading.ThreadHelper.ThreadStart()	Unknown

```

At the bottom is ``Akka.dll!Helios.Concurrency.DedicatedThreadPool.PoolWorker.RunThread`` so this is the helios worker.

At the top is my user code AFTER the Ask. This code SHOULD be run on the ThreadPool, it doesn't make sense to run this on the worker thread, blocking it. As a consumer of Akka, I don't have any real control over that, so it needs to be fixed in Akka.

__Version:__
Akka =1.1.3.

**Comment by 0x53A** [comment]
So I think I got one step further. Here is my investigation in chronological order:

------------------------

The whole program deadlocks if I call .Result, it does not deadlock if I use async / await (makes sense so far).

-------------

One thing I noticed, is that if I call .Result, the message does not even get deserialized:

I have a custom serializer, and put a breakpoint on ``Serializer.FromBinary``.
In the async/await case, this breakpoint gets hit, and then the ask returns.

In the .Result case, the breakpoint doesn't get hit.

-----

I then ran the whole program (with async/await, so that it actually works) under the concurrency visualizer, and noticed the following callstack:

```
Category = Synchronization
SubCategory = Kernel Synch
Delay = 0,109 ms
ntoskrnl.exe!SwapContext+0x31f
ntoskrnl.exe!KiSwapContext+0x76
ntoskrnl.exe!KiSwapThread+0x17c
ntoskrnl.exe!KiCommitThreadWait+0x14f
ntoskrnl.exe!ExpWaitForResource+0x2a8
ntoskrnl.exe!ExAcquireResourceExclusiveLite+0x27e
ntoskrnl.exe!ExEnterCriticalRegionAndAcquireResourceExclusive+0x1b
win32kbase.sys!EnterCrit+0x3c
win32kfull.sys!xxxClientThreadSetup+0x5f
win32kbase.sys!xxxCreateThreadInfo+0x868
win32kbase.sys!UserThreadCallout+0x127
win32kfull.sys!W32pThreadCallout+0x54
win32kbase.sys!W32CalloutDispatch+0x147
ntoskrnl.exe!ExCallCallBack+0x37
ntoskrnl.exe!PsConvertToGuiThread+0xcf
ntoskrnl.exe!KiConvertToGuiThread+0x9
ntoskrnl.exe!KiSystemServiceExitPico+0x113
win32u.dll!NtUserCallOneParam+0x14
user32.dll!MsgWaitForMultipleObjectsEx+0xf9
windowsbase.ni.dll!DomainBoundILStubClass.IL_STUB_PInvoke
windowsbase.ni.dll!MS.Win32.UnsafeNativeMethods.MsgWaitForMultipleObjectsEx
windowsbase.ni.dll!System.Windows.Threading.Dispatcher.RequestBackgroundProcessing
windowsbase.ni.dll!System.Windows.Threading.Dispatcher.CriticalRequestProcessing
windowsbase.ni.dll!System.Windows.Threading.Dispatcher.InvokeAsyncImpl
windowsbase.ni.dll!System.Windows.Threading.Dispatcher.LegacyBeginInvokeImpl
presentationframework.ni.dll!MS.Internal.Data.DataBindEngine.Marshal
presentationframework.ni.dll!MS.Internal.Data.ClrBindingWorker.OnSourcePropertyChanged
windowsbase.ni.dll!System.Windows.WeakEventManager+ListenerList`1[System.__Canon].DeliverEvent
windowsbase.ni.dll!System.ComponentModel.PropertyChangedEventManager.OnPropertyChanged
unknown!0x7ffd26f4f5e3
timdbtoolswpf.dll!Nemetschek.Allready.Logistics.Wpf.ViewModel.TimLogInViewModel.OnPropertyChanged+0x19 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\TimDbToolsWpf\ViewModel\TimLogInViewModel.cs(71, 71)
__timdbtoolswpf.dll!Nemetschek.Allready.Logistics.Wpf.ViewModel.TimLogInViewModel.set_IsLogInEnabled+0x22 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\TimDbToolsWpf\ViewModel\TimLogInViewModel.cs(392, 392)__
timdbtoolswpf.dll!Nemetschek.Allready.Logistics.Wpf.ViewModel.TimLogInViewModel+<StartLogInAsync>d__45.MoveNext+0x106 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\TimDbToolsWpf\ViewModel\TimLogInViewModel.cs(370, 370)
mscorlib.ni.dll!System.Threading.ExecutionContext.RunInternal
mscorlib.ni.dll!System.Threading.ExecutionContext.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c__DisplayClass4_0.<OutputAsyncCausalityEvents>b__0
mscorlib.ni.dll!System.Runtime.CompilerServices.TaskAwaiter+<>c__DisplayClass11_0.<OutputWaitEtwEvents>b__0
mscorlib.ni.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction
mscorlib.ni.dll!System.Threading.Tasks.Task.FinishContinuations
mscorlib.ni.dll!System.Threading.Tasks.Task`1[System.__Canon].TrySetResult
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon].SetResult
timdbtoolswpf.dll!Nemetschek.Allready.Logistics.Wpf.ViewModel.TimLogInViewModel+<LogInAsync>d__44.MoveNext+0xa05 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\TimDbToolsWpf\ViewModel\TimLogInViewModel.cs(61422, 61422)
mscorlib.ni.dll!System.Threading.ExecutionContext.RunInternal
mscorlib.ni.dll!System.Threading.ExecutionContext.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c__DisplayClass4_0.<OutputAsyncCausalityEvents>b__0
mscorlib.ni.dll!System.Runtime.CompilerServices.TaskAwaiter+<>c__DisplayClass11_0.<OutputWaitEtwEvents>b__0
mscorlib.ni.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction
mscorlib.ni.dll!System.Threading.Tasks.Task.FinishContinuations
mscorlib.dll!System.Threading.Tasks.Task`1[System.Guid].TrySetResult
mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.Guid].SetResult
rokku.service.exe!FSharp.Control.Tasks.TaskBuilder+StepStateMachine`1[System.Guid].nextAwaitable+0x33 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\Rokku.Service\TaskBuilder.fs(38, 38)
rokku.service.exe!FSharp.Control.Tasks.TaskBuilder+StepStateMachine`1[System.Guid].System-Runtime-CompilerServices-IAsyncStateMachine-MoveNext+0x1a - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\Rokku.Service\TaskBuilder.fs(58, 58)
mscorlib.ni.dll!System.Threading.ExecutionContext.RunInternal
mscorlib.ni.dll!System.Threading.ExecutionContext.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c__DisplayClass4_0.<OutputAsyncCausalityEvents>b__0
mscorlib.ni.dll!System.Runtime.CompilerServices.TaskAwaiter+<>c__DisplayClass11_0.<OutputWaitEtwEvents>b__0
mscorlib.ni.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction
mscorlib.ni.dll!System.Threading.Tasks.Task.FinishContinuations
mscorlib.ni.dll!System.Threading.Tasks.Task`1[System.__Canon].TrySetResult
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon].SetResult
rokku.service.exe!FSharp.Control.Tasks.TaskBuilder+StepStateMachine`1[System.__Canon].nextAwaitable+0x33 - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\Rokku.Service\TaskBuilder.fs(38, 38)
rokku.service.exe!FSharp.Control.Tasks.TaskBuilder+StepStateMachine`1[System.__Canon].System-Runtime-CompilerServices-IAsyncStateMachine-MoveNext+0x1a - C:\dev\p4\2017PrecastTim\TimPrecast\Sources\Rokku.Service\TaskBuilder.fs(58, 58)
mscorlib.ni.dll!System.Threading.ExecutionContext.RunInternal
mscorlib.ni.dll!System.Threading.ExecutionContext.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run
mscorlib.ni.dll!System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c__DisplayClass4_0.<OutputAsyncCausalityEvents>b__0
mscorlib.ni.dll!System.Runtime.CompilerServices.TaskAwaiter+<>c__DisplayClass11_0.<OutputWaitEtwEvents>b__0
mscorlib.ni.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction
mscorlib.ni.dll!System.Threading.Tasks.Task.FinishContinuations
mscorlib.ni.dll!System.Threading.Tasks.Task`1[System.__Canon].TrySetResult
mscorlib.ni.dll!System.Threading.Tasks.TaskCompletionSource`1[System.__Canon].TrySetResult
akka.dll!Akka.Actor.FutureActorRef.TellInternal+0x30 - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ActorRef.cs(132, 132)
akka.remote.dll!Akka.Remote.EndpointReader.<Reading>b__11_1+0xf1 - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka.Remote\Endpoint.cs(1910, 1910)
anonymously hosted dynamicmethods assembly!dynamicClass.lambda_method
akka.dll!Akka.Tools.MatchHandler.PartialHandlerArgumentsCapture`4[System.__Canon,System.__Canon,System.__Canon,System.__Canon].Handle - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Util\MatchHandler\PartialHandlerArgumentsCapture.cs(143, 143)
akka.dll!Akka.Actor.ReceiveActor.ExecutePartialMessageHandler - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ReceiveActor.cs(78, 78)
akka.dll!Akka.Actor.UntypedActor.Receive+0x7 - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\UntypedActor.cs(27, 27)
akka.dll!Akka.Actor.ActorBase.AroundReceive - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ActorBase.cs(181, 181)
akka.dll!Akka.Actor.ActorCell.ReceiveMessage - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ActorCell.DefaultMessages.cs(179, 179)
akka.dll!Akka.Actor.ActorCell.Invoke+0x5e - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ActorCell.DefaultMessages.cs(83, 83)
akka.dll!Akka.Dispatch.Mailbox.ProcessMailbox+0x19 - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Dispatch\Mailbox.cs(385, 385)
akka.dll!Akka.Dispatch.Mailbox.ProcessMailbox+0x37 - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Dispatch\Mailbox.cs(372, 372)
akka.dll!Akka.Actor.ActorCell.UseThreadContext+0xc - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Actor\ActorCell.cs(399, 399)
akka.dll!Akka.Dispatch.Mailbox.Run+0x1a - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Dispatch\Mailbox.cs(355, 359)
__akka.dll!Helios.Concurrency.DedicatedThreadPool+PoolWorker.RunThread+0x1f - C:\Users\lrieger\Source\Repos\akka.net\src\core\Akka\Helios.Concurrency.DedicatedThreadPool.cs(406, 406)__

```

The exact callstack is irrelevant, but the underlying issue seems to be that my continuations are executed on a HELIOS worker thread, so if I block that thread (which I SHOULD be allowed to), then I block the whole of HELIOS.


I will try again to extract it to a standalone repo ...

**Comment by Danthar** [comment]
This has nothing to do with Akka. It has to do with how the TPL works. And you ran into a classical TPL problem.  :)

Read this: https://blogs.msdn.microsoft.com/pfxteam/2011/01/13/await-and-ui-and-deadlocks-oh-my/

As you already found out in your last example. The first Ask properly uses the await keyword. Which prevents the problem. Because the 2th Ask does not, you get a deadlock for the reasons described in the link.

Another link that provides a more TLDR; version and some alternate solutions: http://stackoverflow.com/questions/17248680/await-works-but-calling-task-result-hangs-deadlocks

**Comment by 0x53A** [comment]
Yeah, I know about the typical deadlock issues.
In my opinion this IS an Akka problem, because Akka incorrectly schedules the continuations. It works correctly if the TaskCompletionSource in FutureActor is created with TaskCreationOptions.RunContinuationsAsynchronously.

The issue is that after the first Ask, the continuation is executed on the helios thread, not on the threadpool. As a consumer of Akka, I don't have any control over that (as far as I know), so it needs to be fixed in Akka.

**Comment by alfredherr** [comment]
I think I just ran into this issue as well. I'm on Akka cluster 1.3.1, on DotNetCore 2.0 on MacOS (and docker/ubuntu). 

It's a POC of using Akka with an DotNetCore ASP API. I'm registering the actor system in the StartUp class.

```C#
 public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        public void ConfigureServices(IServiceCollection services)
        {

            /* register the actor system */
            ActorSystemRefs.ActorSystem = ActorSystem.Create("demo-system", getConfiguration());
            ActorMonitoringExtension.RegisterMonitor(ActorSystemRefs.ActorSystem, new ActorStatsDMonitor(host: "localhost", port: 8125, prefix: "akka-demo"));
            SystemActors
               .AccountSupervisor =
                  ActorSystemRefs
                  .ActorSystem
                  .ActorOf(Props.Create<AccountActorSupervisor>(), name: "demo-supervisor");

           /* this fails with an Ask() and this Tell never makes it - the supervisor never gets the message*/
            SystemActors.AccountSupervisor.Tell(new AboutMe("hello Supervisor"));
           
            /* then add MVC */
            services.AddMvc();


        }
```

here is a link [https://github.com/alfredherr/AkkaDotNetCoreDocker/blob/f6488244024a30fe4db53f12e9b14b342b00e2d6/WebClient/Startup.cs#L52]

The really strange thing is that it was working for a while -- then, for some reason, it's throwing that error 
```C#
System.AggregateException: One or more errors occurred. ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.

```

I am at a loss here. I'm not a C# expert, and from what I can tell, I can't use Akka with ASP.NET (!?)

I'm trying to get our financial services organization to consider Akka for a system rewrite. Any help would be really appreciated.

**Comment by Horusiath** [comment]
@alfredherr So far I'm not sure, how to solve this issue. However in my case workaround was wrapping problematic `Ask`/`Tell` with a `Task.Run(() => ...)`.

**Comment by Aaronontheweb** [comment]
@alfredherr I don't think the issue you're having is related to this one, since you indicated that the message sent via `Tell` didn't arrive... Could you double check the actor code and ensure that it's actually handling the message correctly?

**Comment by Aaronontheweb** [comment]
@alfredherr also, feel free to IM me on our Gitter channel if you want to PM a private sample or something

**Comment by Aaronontheweb** [comment]
Resolved via #3120
